boards: platform: add virtual board

hakehuang opened the pull request:
virtual board is to use the master platfrom to
test soc derivatives

the generated files is just for demoing purpose, will be removed when the idea is accepted

RFC: https://github.com/zephyrproject-rtos/zephyr/issues/102733

nordicjm reviewed (changes requested):
No way, put this in your own private repo, do not litter the zepo repo with test files

hakehuang commented:
> No way, put this in your own private repo, do not litter the zepo repo with test files

sorry, this is only a draft, the generated boards file will be removed, and only the methodology need review. I put the generated file here for easy to understand, I plan to remove it full reviewed

hakehuang commented on samples/boards/nxp/virtual_board/README.md:
so can I move it the /boards/nxp/virtual_boards, instead?

hakehuang commented:
@butok, @DerekSnell  please review. this is an idea to validate all possible derived parts vs master platform for feature completeness check.

JarmouniA commented on boards/virtual_board/README.md:
convert it to '.rst'

hakehuang commented:
> Can we use the Nordic approach for the derivative testing? No need to create "virtual" boards.
> For example, nrf52dk is used to test subset SOC derivatives:

@butok, I am using the the same way, the nrf52dk is select as master platforms, and generate boards for different supporting parts. I just add script to automatic do this. the generated folder will be removed(just to demo the script result) if this approach is ok.

hakehuang committed:
boards: add virtual boards support

using a json file to control the relationsjip with derivated soc
and generate virtualboards
expand targets and improve DTS include handling

Signed-off-by: Hake Huang <hake.huang@nxp.com>

butok commented:
Hi @hakehuang 

The following commits/fixes are independent and required not only for the Virtual board generation. So they should be added as separate PRs: 
- soc: imx6sx : code change to support m4 core
- modules: hal_nxp: mcux: always enable lpcflexcomm component when used
- soc: nxp: fix MCX* CPU feature selects and add i.MXRT118x CM7 MPU
- drivers: clock_control: mcux: fix RW6xx syscon clock rates

mmahadevan108 commented on boards/nxp/frdm_mcxw72/frdm_mcxw72_mcxw727c_cpu1.dts:
Can this commit be moved to a separate PR?

hakehuang replied:
sure, I will do this.

mmahadevan108 commented on boards/nxp/mcxw72_evk/Kconfig.mcxw72_evk:
This commit says there is no need need for CPU variant for this platform
https://github.com/zephyrproject-rtos/zephyr/pull/95684

mmahadevan108 commented on boards/nxp/vmu_rt1170/vmu_rt1170_mimxrt1176_cm4.dts:
Can be a separate PR I think

hakehuang replied:
sure

hakehuang commented on boards/nxp/imx95_evk_15x15/imx95_evk_15x15_mimx9596_m7.dts:
sure I will do this
